File: ch10/spring-boot-example/src/main/java/com/packtpub/hands/on/reactive/kotlin/chapter10/springbootexample/ExampleController.java
package com.packtpub.hands.on.reactive.kotlin.chapter10.springbootexample;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ExampleController {
  private final CustomerInterface customerInterface;

  public ExampleController(CustomerInterface customerInterface) {
    this.customerInterface = customerInterface;
  }

  @GetMapping("/customer/{number}")
  Customer hello(@PathVariable int number) {
    return customerInterface.getCustomer(number);
  }
}

File: ch11/spring-boot-kotlin-example/src/main/kotlin/com/packtpub/hands/on/reactive/kotlin/chapter11/springbootkotlinexample/ExampleController.kt
package com.packtpub.hands.on.reactive.kotlin.chapter11.springbootkotlinexample

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class ExampleController(val customerInterface: CustomerInterface) {
  @GetMapping("/customer/{id}")
  fun getCustomer(@PathVariable id: Int) = customerInterface.getCustomer(id)
}
